components/remote_settings/src/remote_settings.udl

```diff
@@ -73,3 +77,50 @@ interface RemoteSettings {
     [Throws=RemoteSettingsError]
     void download_attachment_to_path(string attachment_id, string path);
 };
+
+/// Stores application-level configuration data and constructs clients
+interface RemoteSettingsService {
+    constructor(RemoteSettingsConfig2 config);
+
+    /// Construct a new client with the configuration passed to the constructor
+    [Throws=RemoteSettingsError]
+    RemoteSettingsClient make_client(string collection_name);
+};
+
+/// Configuration for RemoteSettingsService
+///
+/// Once all clients have moved to the new API, we can remove the old RemoteSettingsConfig and
+/// rename this.
+dictionary RemoteSettingsConfig2 {
+    /// Server to use
+    RemoteSettingsServer server;
+    /// Directory to store data in.  Must be writable by the client.
+    string storage_dir;
+    /// Override bucket name ("main" by default, "preview" for preview collections)
+    string? bucket_name = null;
+};
+
+/// New remote settings client
+///
+/// This is a WIP matches the semantics of the official client more closely.  Once this is complete,
+/// we may decide to deprecate and remote the RemoteSettings class.
+interface RemoteSettingsClient {
+    /// Get the current set of records.
+    ///
+    /// If no records have been downloaded, this will return null.  Use `sync` to fetch the current
+    /// records from the server.
+    [Throws=RemoteSettingsError]
+    sequence<RemoteSettingsRecord>? get();
+
+    /// Sync records with the server.
+    [Throws=RemoteSettingsError]
+    void sync();
```

Comment:
Are we going to also periodically check with the server for updates? If so, it might be worth making clear that this is only needed in the force-sync case.

Reply:
This is a great question.  I think the periodic syncing should be driven by the embedding application, since they have a closer integration with the OS and can create background workers, etc.  However, it seems annoying to make each remote settings client consumer have to call `sync()` for their client.

How about this:
- We rename this method to `force_sync()` and documentation about that.
- We add a `RemoteSettingsService::sync()` method, which is what the embedding application will call.  That updates the remote settings data for all the clients that it knows about (or something similar, I'm not sure exactly how this should work).